fmlloader/src/main/java/net/minecraftforge/fml/loading/moddiscovery/ModInfo.java

```diff
@@ -86,10 +86,16 @@ public ModInfo(final ModFileInfo owningFile, final IConfigurable config)
                 .collect(Collectors.toList()))
                 .orElse(Collections.emptyList());
 
-        this.features = ownFile.map(mfi -> mfi.<Map<String, String>>getConfigElement("features", this.modId)
+        this.features = ownFile.map(mfi -> mfi.<Map<String, Object>>getConfigElement("features", this.modId)
                 .stream()
                 .flatMap(m->m.entrySet().stream())
-                .map(e->new ForgeFeature.Bound(e.getKey(), e.getValue(), this))
+                .map(e->{
+                    if (e.getValue() instanceof String val) {
+                        return new ForgeFeature.Bound(e.getKey(), val, this);
+                    } else {
+                        throw new InvalidModFileException("Invalid feature bound {"+e.getValue()+"} for key {"+e.getKey()+"} only strings are accepted", this.owningFile);
+                    }
+                })
                 .collect(Collectors.toList())).orElse(Collections.emptyList());
```

Comment:
This (and the other spots in this file) should probably be switched to `.toList()` rather than `.collect(Collectors.toList())` unless it is intentional that the features etc are mutable, in which case the `orElse` should return a mutable list

Reply:
I debated that, and yes, they should all be immutable collections - but I didn't want to inflate the size of the changeset.